### PR TITLE
[IMP] project: Copy value from date_deadline to date_end.

### PR DESCRIPTION
**Pull request by sergio-teruel**

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Seems reasonable to propagate the value. Maybe an smarter solution can be to look if the value is the same on both fields before the change (or the other is empty), and if it's the case, then make the change, and respect the different data if not.

**Comment by rafaelbn** [comment]
In v9 Odoo unify this 2 fields in 1. :+1:

**Comment by rafaelbn** [comment]
@mart-e @beledouxdenis could you consider this or close it please?

**Comment by beledouxdenis** [comment]
Could you please stop mentioning me in each pull requests you want to merge ? Pretty please ?

**Comment by rafaelbn** [comment]
OK, sorry.

**Comment by rafaelbn** [comment]
@pedrobaeza it looks this PR will be not accepted from Odoo, can we close it and finish it in OCB? what do you think @sergio-incaser ?

**Comment by dreispt** [comment]
I'm not sure this should go in OCB. But make a PR there and let's see other opinions.

**Comment by rafaelbn** [comment]
@dreispt this PR https://github.com/OCA/OCB/pull/328 was merged in OCB at 10:55 and you comment was at 18:11. This PR still opened waiting to be merge by Odoo

File: addons/project/project.py [hunk 1]
```diff
@@ -686,6 +686,12 @@ def onchange_user_id(self, cr, uid, ids, user_id, context=None):
             vals['date_start'] = fields.datetime.now()
         return {'value': vals}
 
+    def onchange_date_deadline(
```
Comment (mart-e) [inline addons/project/project.py]: Put everything on one line, we [don't follow](https://www.odoo.com/documentation/9.0/reference/guidelines.html#pep8-options) the 80 char guidelines (here it actually reduces the readability IMHO)

File: addons/project/project.py [hunk 2]
```diff
@@ -686,6 +686,12 @@ def onchange_user_id(self, cr, uid, ids, user_id, context=None):
...
+            self, cr, uid, ids, date_end, date_deadline, context=None):
+        if not date_end or (date_end[:10] == self.browse(
```
Comment (mart-e) [inline addons/project/project.py]: browsing a record in an onchange will not work for records not created yet, I don't see the point of checking the value in db anyway.

File: addons/project/project.py [hunk 3]
```diff
@@ -686,6 +686,12 @@ def onchange_user_id(self, cr, uid, ids, user_id, context=None):
...
+                cr, uid, ids, context=context).date_deadline):
```
Comment (mart-e) [inline addons/project/project.py]: So if the dates are already the same, you modify it anyway without the time information? Why?

**Comment by sergio-incaser** [comment]
@mart-e The purpose of this module is fill date_end field with date_deadline value when the logic allows:
* If date_end gap.
* If date_end is equal to date_deadline before changes (in this comparison we must discard date_end time because it is datetime and date_deadline field is date)

**Comment by sergio-incaser** [comment]
cc @rafaelbn @pedrobaeza @dreispt

**Comment by pedrobaeza** [comment]
@sergio-incaser, when I proposed the comparison, I didn't realize that this was an onchange, so using browse is not advisable because it will only contain the saved content. That's why you pass the other field value as onchange argument.

**Comment by rafaelbn** [comment]
About this PR, I see that @sergio-incaser change as suggested by @mart-e . But @pedrobaeza @sergio-incaser  we must make a PR in OCB to add this improvements as it was merged in https://github.com/OCA/OCB/pull/328

**Comment by mart-e** [comment]
Sorry but as said I still don't see the point to compare with the value in db. It seems to be checking the previous value but that's not the case (not working for new records, neither if I change twice the value).

Anyway this is an improvement, [not a bug fix](https://github.com/odoo/odoo/wiki/Contributing#what-does-stable-mean). I would have accepted it in upper version (if corrected) but the field date_end has been removed.

**Comment by rafaelbn** [comment]
Hi @mart-e , but Odoo v9 doesn't have this two fields, as Odoo v9 no have gannt. Maybe you can consider?

**Comment by rafaelbn** [comment]
Please @sergio-incaser apply changes done in OCB. Thanks.

**Pull request closed without merging**